app/src/main/java/tech/relaycorp/gateway/domain/courier/ProcessCargo.kt

```diff
@@ -49,7 +49,8 @@ class ProcessCargo
         val parcel = when (
             val result = storeParcel.store(parcelData, RecipientLocation.LocalEndpoint)
         ) {
-            is StoreParcel.Result.MalformedParcel -> return
+            is StoreParcel.Result.MalformedParcel,
+            is StoreParcel.Result.DuplicatedParcel -> return
```

Comment:
I think getting a duplicated parcel shouldn't be logged as a warning that says "Invalid parcel received" because that's misleading. I think it should either not be logged (like in the success case) or logged with a debug/info message that says something like "Ignoring previously-processed parcel".

Reply:
You probably misread, cause right now it's not logging anything like the MalformedParcel :)

Reply:
haha, I did! :smile: